src/util/helperUtil.ts

```diff
@@ -80,3 +81,53 @@ export function clearDataSavedInGlobalState(context: vscode.ExtensionContext) {
 	context.globalState.update('selectedProject', undefined);
 }
 
+/**
+ * Checks if sibling files exist in the parent directory of the 'target' or 'build' folder, excluding the current directory.
+ * @param excludeDir - The directory where the file is generated, which will be excluded from the search.
+ * @param fileType - The type of file being watched.
+ * @returns - true if a sibling file is found, otherwise false.
+ */
+export async function checkSiblingFilesInTargetOrBuildParent(excludeDir: string, fileType: string): Promise<boolean> {
+
+	try {
+
+        // Get parentDir based on whether 'target' or 'build' exists in the path
+        const pathParts = excludeDir.split(path.sep);
+        const targetIndex = pathParts.lastIndexOf('target');
+        const buildIndex = pathParts.lastIndexOf('build');
+        
+        const parentDir = targetIndex !== -1 ? pathParts.slice(0, targetIndex).join(path.sep)
```

Comment:
Can you add some comments to explain this part of the code? The name `parentDir` does not seem accurate here, though I may not be following. Parent dir of what exactly?